diff --git a/testprogs/Makefile.in b/testprogs/Makefile.in
index f5f65ee7..7757b059 100644
--- a/testprogs/Makefile.in
+++ b/testprogs/Makefile.in
@@ -80,7 +80,7 @@ SRC = @VALGRINDTEST_SRC@ \
 	capturetest.c \
 	filtertest.c \
 	findalldevstest-perf.c \
-	findalldevstest.c \
+	# findalldevstest.c \
 	opentest.c \
 	nonblocktest.c \
 	reactivatetest.c \
@@ -102,59 +102,63 @@ all: $(TESTS)
 
 activatetest: $(srcdir)/activatetest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/activatetest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 capturetest: $(srcdir)/capturetest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/capturetest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 can_set_rfmon_test: $(srcdir)/can_set_rfmon_test.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ \
 	    $(srcdir)/can_set_rfmon_test.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 filtertest: $(srcdir)/filtertest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/filtertest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 findalldevstest: $(srcdir)/findalldevstest.c ../libpcap.a
-	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ \
-	    $(srcdir)/findalldevstest.c \
-	    ../libpcap.a $(LIBS)
+	ifeq($(shell uname),"OS/390") ; then
+		@echo "** $@ won't be built for z/OS **"
+	else
+		$(CC) $(FULL_CFLAGS) -I. -L. -o $@ \
+			$(srcdir)/findalldevstest.c \
+			$(LDFLAGS) ../libpcap.a $(LIBS)
+	endif
 
 findalldevstest-perf: $(srcdir)/findalldevstest-perf.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ \
 	    $(srcdir)/findalldevstest-perf.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 opentest: $(srcdir)/opentest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/opentest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 nonblocktest: $(srcdir)/nonblocktest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/nonblocktest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 reactivatetest: $(srcdir)/reactivatetest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ \
-	    $(srcdir)/reactivatetest.c ../libpcap.a $(LIBS)
+	    $(srcdir)/reactivatetest.c $(LDFLAGS) ../libpcap.a $(LIBS)
 
 selpolltest: $(srcdir)/selpolltest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/selpolltest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 threadsignaltest: $(srcdir)/threadsignaltest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ \
 	    $(srcdir)/threadsignaltest.c \
-	    ../libpcap.a $(LIBS) $(PTHREAD_LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS) $(PTHREAD_LIBS)
 
 valgrindtest: $(srcdir)/valgrindtest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/valgrindtest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 writecaptest: $(srcdir)/writecaptest.c ../libpcap.a
 	$(CC) $(FULL_CFLAGS) -I. -L. -o $@ $(srcdir)/writecaptest.c \
-	    ../libpcap.a $(LIBS)
+	    $(LDFLAGS) ../libpcap.a $(LIBS)
 
 clean:
 	rm -f $(CLEANFILES)
