diff --git a/etherent.c b/etherent.c
index f367c9dd..150fc0dc 100644
--- a/etherent.c
+++ b/etherent.c
@@ -38,6 +38,10 @@
 #include "os-proto.h"
 #endif
 
+#ifdef __MVS__
+#include <zos-tls.h>
+#endif
+
 static inline int skip_space(FILE *);
 static inline int skip_line(FILE *);
 
@@ -76,7 +80,15 @@ pcap_next_etherent(FILE *fp)
 	u_char d;
 	char *bp;
 	size_t namesize;
+#ifdef __MVS__
+	static tls_t tls_e;
+	ssize_t size_e = sizeof(struct pcap_etherent);
+	tls_e.sz = size_e;
+	struct pcap_etherent *e_ptr = (struct pcap_etherent *)__tlsValue(&tls_e);
+	#define e (*e_ptr)
+#else
 	static thread_local struct pcap_etherent e;
+#endif
 
 	memset((char *)&e, 0, sizeof(e));
 	for (;;) {
